base/iostream.jl

```diff
@@ -225,8 +225,18 @@ function readuntil(s::IOStream, delim::UInt8)
     ccall(:jl_readuntil, Array{UInt8,1}, (Ptr{Void}, UInt8, UInt8), s.ios, delim, 0)
 end
 
-function readline(s::IOStream)
-    ccall(:jl_readuntil, Ref{String}, (Ptr{Void}, UInt8, UInt8), s.ios, '\n', 1)
+function readline(s::IOStream, chomp::Bool=false)
+    chomp || return ccall(:jl_readuntil, Ref{String}, (Ptr{Void}, UInt8, UInt8), s.ios, '\n', 1)
+
+    s = ccall(:jl_readuntil, Ref{String}, (Ptr{Void}, UInt8, UInt8), s.ios, '\n', 1)
+    i = endof(s)
+    if i < 1 || codeunit(s,i) != 0x0a
+        return s[1:i]
```

Comment:
Unfortunately, `String(::Vector)` is no longer copy-free unless the vector is allocated with the (undocumented) `Base.StringVector`:
```jl
julia> x = Base.StringVector(2); x[1] = 'a'; x[2] = 'b'
'b': ASCII/Unicode U+0062 (category Ll: Letter, lowercase)

julia> x = Base.StringVector(2); x[1] = 'a'; x[2] = 'b'; s = String(x)
"ab"

julia> x[1] = 'c'
'c': ASCII/Unicode U+0063 (category Ll: Letter, lowercase)

julia> s
"cb"
```